subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultProviderFactory.java

```diff
@@ -69,9 +67,9 @@ public DefaultProviderFactory(FileOperations fileOperations) {
             ((PropertyState<Double>) propertyState).set(Double.valueOf(0));
         } else if (clazz == Character.class) {
             ((PropertyState<Character>) propertyState).set(new Character('\0'));
-        } else if (clazz == FileCollection.class || clazz == ConfigurableFileCollection.class) {
+        } else if (clazz == ConfigurableFileCollection.class) {
```

Comment:
OK, let me summarize what I understand. At the moment we do not want to give people the option to create a `PropertyState` for any `FileCollection` simply because the next story addresses the concern in the intended way and we don't want to promote a usage pattern that will soon be outdated plus become a breaking change.

That in turn would mean that my test cases should not use any `PropertyState<FileCollection>` at the moment. It also means that `FileCollection` types cannot be used for mapping extension properties to task properties until the next story is implemented (FileCollection is-a Provider). A `Provider<ConfigurableFileCollection>` as a replacement for `PropertyState<FileCollection>` won't work for properly mapping it as you'd change the reference to the original, immutable `Provider` when using the extension  to set a value from the build script.

Is that a correct assessment? Ultimately, that means that we probably shouldn't even promote the concept in the release notes until we actually implement the next story because most likely users will want to map `FileCollection`s.